File: API/models/etb.ts
const mongooseEtb = require('mongoose')

/**
 * @openapi
 * components:
 *  schemas:
 *    EtbSchema:
 *      type: object
 *      required:
 *        - nameEtb
 *        - lieu
 *        - prix
 *        - capMin
 *        - capMax
 *        - heureO
 *        - heureF
 *        - type
 *        - ownerId
 *        - coordonesGPS
 *        - description
 *      properties:
 *        nameEtb:
 *          type: String
 *        lieu:
 *          type: String
 *        heureO:
 *          type: Date
 *        heuerF:
 *          type: Date
 *        capMin:
 *          type: Number
 *        capMax:
 *          type: Number
 *        description:
 *          type: String
 *        prix:
 *          type: String
 *
 */
const etbSchema = new mongooseEtb.Schema({
    nomEtablissement: {type:String, required:true, unique:true },
    prix: {type:String, required:true},
    lieu: {type:String, required:true},
    heureOuverture: {type:Date, required:true},
    heureFermeture: {type:Date, required:true},
    type: {type: String, required:true},
    userId: {type: mongooseEtb.ObjectId, required:true},
    images: [{type: String}],
    capaciteMax: {type:Number},
    capaciteMin: {type:Number},
    location: {
        type: {type: String},
        coordinates: []
    },
    description: {type: String}

})


etbSchema.index({location:"2dsphere"})
const etbModel = mongooseEtb.model("etablissements",etbSchema)
module.exports = {Etb: etbModel}

File: app_web/lib/models/etb.dart
import 'dart:convert';

class Etablissement{

  final String? userId;
  final String? id;
  final String type;
  final String lieu;
  final String heureOuverture;
  final String heureFermeture;
  final int capaciteMax;
  final int capaciteMin;
  final String nameEtb;
  final List<String> images;
  final String prix;
  final String description;

  Etablissement({
    this.userId,
    this.id, 
    required this.type, 
    required this.lieu, 
    required this.heureOuverture,
    required this.heureFermeture,
    required this.capaciteMax,
    required this.capaciteMin,
    required this.nameEtb,
    required this.images,
    required this.prix,
    required this.description
  });

  Map<String, dynamic> toMap() {
      
      return {
        'userId': userId,
        '_id': id,
        'heureOuverture': heureOuverture,
        'type': type,
        'lieu': lieu,
        'heureFermeture': heureFermeture,
        'capaciteMax': capaciteMax,
        'capaciteMin': capaciteMin,
        'nameEtb': nameEtb,
        'images': images,
        'prix': prix,
        'description': description
      };
    }
  factory Etablissement.fromMap(Map<String, dynamic> map){
    return Etablissement(
      userId: map['userId'] ?? '',
      id: map['_id'] ?? '', 
      type: map['type'] ?? '', 
      lieu: map['lieu'] ?? '', 
      heureOuverture: map['heureOuverture'] ?? '',
      heureFermeture: map['heureFermeture'] ?? '',
      capaciteMax: map['capaciteMax'] ?? '',
      capaciteMin: map['capaciteMin'] ?? '',
      nameEtb: map['nomEtablissement'] ?? '',
      images: List<String>.from(map['images']),
      prix: map['prix'] ?? '',
      description: map['description'] ?? ''
    );
  }

  String toJson() => json.encode(toMap());
  factory Etablissement.fromJson(String source) => Etablissement.fromMap(json.decode(source));

}